src/IO/BrotliWriteBuffer.cpp

```diff
@@ -49,9 +49,6 @@ BrotliWriteBuffer::BrotliWriteBuffer(std::unique_ptr<WriteBuffer> out_, int comp
 
 BrotliWriteBuffer::~BrotliWriteBuffer()
 {
-    /// FIXME move final flush into the caller
-    MemoryTracker::LockExceptionInThread lock;
-    finish();
```

Comment:
But why do you removed `finish()`? What if someone will forget to call `finalize()`?